tests/bot/cogs/moderation/test_utils.py

```diff
@@ -0,0 +1,477 @@
+                    "hidden": False,
+                    "reason": "Test Ban",
+                    "type": "note",
+                    "user": self.member.id,
+                    "active": True
+                }
+            },
+            {
+                "args": (self.ctx, self.member, "mute", "Test Ban"),
+                "expected_output": None,
+                "raised_error": ResponseCodeError(AsyncMock(status=400), {'user': 1234}),
+                "payload": {
+                    "actor": self.ctx.message.author.id,
+                    "hidden": False,
+                    "reason": "Test Ban",
+                    "type": "mute",
+                    "user": self.member.id,
+                    "active": True
+                }
+            },
+            {
+                "args": (self.ctx, self.member, "ban", "Test Ban", now, True, False),
+                "expected_output": [
+                    {
+                        "id": 1,
+                        "inserted_at": "2018-11-22T07:24:06.132307Z",
+                        "expires_at": "5018-11-20T15:52:00Z",
+                        "active": True,
+                        "user": 1234,
+                        "actor": 1234,
+                        "type": "ban",
+                        "reason": "Test Ban",
+                        "hidden": False
+                    }
+                ],
+                "raised_error": None,
+                "payload": {
+                    "actor": self.ctx.message.author.id,
+                    "hidden": True,
+                    "reason": "Test Ban",
+                    "type": "ban",
+                    "user": self.member.id,
+                    "active": False,
+                    "expires_at": now.isoformat()
+                }
+            },
+        ]
+
+        for case in test_cases:
+            args = case["args"]
+            expected = case["expected_output"]
+            raised = case["raised_error"]
+            payload = case["payload"]
+
+            with self.subTest(args=args, expected=expected, raised=raised, payload=payload):
+                self.ctx.bot.api_client.post.reset_mock(side_effect=True)
+                post_user_mock.reset_mock()
+
+                if raised:
+                    self.ctx.bot.api_client.post.side_effect = raised
+
+                post_user_mock.return_value = "foo"
+
+                self.ctx.bot.api_client.post.return_value = expected
+
+                result = await post_infraction(*args)
+
+                self.assertEqual(result, expected)
+
+                if not raised:
+                    self.bot.api_client.post.assert_awaited_once_with("bot/infractions", json=payload)
+
+                if hasattr(raised, "status") and hasattr(raised, "response_json"):
+                    if raised.status == 400 and "user" in raised.response_json:
+                        post_user_mock.assert_awaited_once_with(args[0], args[1])
```

Comment:
This isn't testing the false branch in which an error message should be sent to the context.